src/vm/analysis/trait_checker/tests.rs

```diff
@@ -840,3 +840,55 @@ fn test_good_call_2_with_trait() {
     }).unwrap();
 }
 
+#[test]
+fn test_contract_of_good() {
+    let contract_defining_trait =
+        "(define-trait trait-1 (
+            (get-1 (uint) (response uint uint))))";
+    let dispatching_contract =
+        "(use-trait trait-2 .defun.trait-1)
+        (define-public (wrapped-get-1 (contract <trait-2>))
+            (ok (contract-of contract)))";
+    let def_contract_id = QualifiedContractIdentifier::local("defun").unwrap();
+    let disp_contract_id = QualifiedContractIdentifier::local("dispatch").unwrap();
+    let mut c1 = parse(&def_contract_id, contract_defining_trait).unwrap();
+    let mut c2 = parse(&disp_contract_id, dispatching_contract).unwrap();
+    let mut marf = MemoryBackingStore::new();
+    let mut db = marf.as_analysis_db();
+
+    db.execute(|db| {
+        type_check(&def_contract_id, &mut c1, db, true).unwrap();
+        type_check(&disp_contract_id, &mut c2, db, true)
+    }).unwrap();
+
+}
+
+#[test]
+fn test_contract_of_wrong_type() {
+    let contract_defining_trait =
+        "(define-trait trait-1 (
+            (get-1 (uint) (response uint uint))))";
+    let dispatching_contract =
+        "(use-trait trait-2 .defun.trait-1)
+        (define-public (wrapped-get-1 (contract principal))
```

Comment:
Can you expand this test to verify that the `contract-of` built-in fails for other types as well, besides `principal`?

Reply:
addressed in c15c1af8eee54c228668b91b56570ddf6007c0dc